File: app/src/main/java/org/deiverbum/app/model/NuncDimitis.java
package org.deiverbum.app.model;

import android.text.SpannableStringBuilder;

import org.deiverbum.app.utils.Utils;

public class NuncDimitis extends Salmo{

    //private String antifonaEntity;
    private String texto;

    public NuncDimitis() {
    }

    public String getTexto() {
        return Utils.getFormato(texto);
    }

    public void setTexto(String texto) {
        this.texto = texto;
    }

    public SpannableStringBuilder getAntifonaSpan(int timeID) {
        SpannableStringBuilder ssb = new SpannableStringBuilder("");
        ssb.append(Utils.toRed("Ant. "));
        ssb.append(Utils.replaceByTime(antifona, timeID));
        return ssb;
    }

    public String getAntifonaForRead() {
        return antifona;
    }



    public SpannableStringBuilder getHeader() {
        return Utils.formatTitle("CÁNTICO EVANGÉLICO");
    }

    public String getHeaderForRead() {
        return "CÁNTICO EVANGÉLICO";
    }

    public SpannableStringBuilder getAll(int idTiempo) {
        SpannableStringBuilder sb=new SpannableStringBuilder();
        sb.append(getHeader());
        sb.append(Utils.LS2);
        sb.append(getAntifonaSpan(idTiempo));
        sb.append(Utils.LS2);
        sb.append(Utils.fromHtml(getTexto()));
        sb.append(Utils.LS2);
        sb.append(getFinSalmo());
        sb.append(Utils.LS2);
        sb.append(getAntifonaSpan(idTiempo));
        return sb;
    }

    public SpannableStringBuilder getAllForRead() {
        SpannableStringBuilder sb=new SpannableStringBuilder();
        sb.append(getHeaderForRead());
        sb.append(".");
        sb.append(getAntifonaForRead());
        sb.append(Utils.fromHtml(getTexto()));
        sb.append(getFinSalmoForRead());
        sb.append(getAntifonaForRead());
        return sb;
    }
}

File: app/src/main/java/org/deiverbum/app/model/NuncDimitis.kt
package org.deiverbum.app.model

import android.text.SpannableStringBuilder
import org.deiverbum.app.util.Constants
import org.deiverbum.app.util.Utils

class NuncDimitis : LHPsalm() {
    //private String antiphonEntity;
    var texto: String? = null
        get() = Utils.getFormato(field)

    private fun getAntifonaSpan(timeID: Int): SpannableStringBuilder {
        val ssb = SpannableStringBuilder("")
        ssb.append(Utils.toRed("Ant. "))
        ssb.append(Utils.replaceByTime(antiphon, timeID))
        return ssb
    }

    override val antifonaForRead: String
        get() = antiphon!!
    val header: SpannableStringBuilder
        get() = Utils.formatTitle(Constants.TITLE_GOSPEL_CANTICLE)
    private val headerForRead: String
        get() = Utils.pointAtEnd(Constants.TITLE_GOSPEL_CANTICLE)

    fun getAll(idTiempo: Int): SpannableStringBuilder {
        val sb = SpannableStringBuilder()
        sb.append(header)
        sb.append(Utils.LS2)
        sb.append(getAntifonaSpan(idTiempo))
        sb.append(Utils.LS2)
        sb.append(Utils.fromHtml(texto))
        sb.append(Utils.LS2)
        sb.append(finSalmo)
        sb.append(Utils.LS2)
        sb.append(getAntifonaSpan(idTiempo))
        return sb
    }

    val allForRead: SpannableStringBuilder
        get() {
            val sb = SpannableStringBuilder()
            sb.append(headerForRead)
            sb.append(".")
            sb.append(antifonaForRead)
            sb.append(Utils.fromHtml(texto))
            sb.append(finSalmoForRead)
            sb.append(antifonaForRead)
            return sb
        }
}